### PR TITLE
Adjust codecov: only patch statuses, wait for CI to pass

### LINKED ISSUE #6994: [RFC] consider droping codecov
With codecov i observe a lot of failures/issues/false positives.

So i would appreciate if we found and switched to a more stable alternative.

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #6994

### CODE REVIEW

**Comment by nicoddemus** [comment]
I expect that only `codecov/patch` will appear once this gets merged to master.

**Comment by gnikonorov** [comment]
as per https://docs.codecov.io/docs/commit-status#excluding-tests-example, do we need to specify project as false?

```
coverage:
  status:
    project:
      default: false
```

File: codecov.yml [hunk 1]
```diff
@@ -1 +1,7 @@
-comment: off
+# reference: https://docs.codecov.io/docs/codecovyml-reference
+codecov:
+  require_ci_to_pass: true
```
Comment (bluetech) [inline codecov.yml:3]: According to the reference this is the default, so maybe want to omit.

**Comment by bluetech** [comment]
I think @gnikonorov is correct, though this section https://docs.codecov.io/docs/commit-status#disabling-a-status rather suggests

```yml
coverage:
  status:
    project: off
```

**Comment by nicoddemus** [comment]
You folks are right! Funny thing, I was thinking exactly that this was the correct way in the shower this morning. 😆

**Comment by nicoddemus** [comment]
Backport: https://github.com/pytest-dev/pytest/pull/7299

**Pull request merged**